lp-core-platform/lp-cp-apis/src/main/java/eu/learnpad/core/impl/cw/XwikiController.java

```diff
@@ -20,30 +20,91 @@
 package eu.learnpad.core.impl.cw;
 
 import javax.inject.Named;
+import javax.inject.Singleton;
 import javax.ws.rs.Path;
 
 import org.xwiki.component.annotation.Component;
+import org.xwiki.component.phase.Initializable;
+import org.xwiki.component.phase.InitializationException;
 import org.xwiki.rest.XWikiRestComponent;
 
 import eu.learnpad.core.rest.RestResource;
 import eu.learnpad.core.rest.XWikiRestUtils;
 import eu.learnpad.cw.Controller;
 import eu.learnpad.exception.LpRestException;
+import eu.learnpad.cw.BridgeInterface;
 
+/*
+ * It is not clear yet who is responsible for the instantiation
+ * of this class. From what I read from Jean it is supposed to be done
+ * automatically when registering the Java component into the XWiki Platform.
+ * Thus I do not know how may instances we may actually have. Thus, for the
+ * moment I marked it as Singleton.
+ *  
+ */
 @Component
+@Singleton
 @Named("eu.learnpad.core.impl.cw.XwikiController")
 @Path("/learnpad/cw/corefacade")
-public class XwikiController extends Controller implements XWikiRestComponent {
+public class XwikiController extends Controller implements XWikiRestComponent, Initializable{
 
-	public XwikiController() {
-		this(false);
-	}
+    /** Set to true once the inherited BridgeInterface has been initialized. */
+    private boolean initialized = false;	
+
+    /*
+     * Note that in this solution the Controllers do not interact
+     * each-others, but each controller directly invokes the BridgesInterfaces
+     * (from the other controllers) it needs. This is not actually what was
+     * originally planned, thus in the future it may change.
+     *
+     * Also, not sure if this is the correct way to proceed.
+     * I would like to decide in a configuration file
+     * the implementation to bind, and not into the source
+     * code. In fact, this second case implies to rebuild the
+     * whole platform at each change.	
+     */
```

Comment:
Yes, this is true, but I would prefer that than nothing. So please consider the simplest solution for now and a configuration file is a possible evolution in the future.